library/cc/engine_builder.cc

```diff
@@ -54,6 +54,25 @@ EngineBuilder::addDnsPreresolveHostnames(const std::string& dns_preresolve_hostn
   return *this;
 }
 
+EngineBuilder& EngineBuilder::addH2ConnectionKeepaliveIntervalSeconds(
+    int h2_connection_keepalive_interval_seconds) {
+  this->h2_connection_keepalive_interval_seconds_ = h2_connection_keepalive_interval_seconds;
+  return *this;
+}
+
+EngineBuilder& EngineBuilder::addH2ConnectionKeepaliveIdleIntervalMilliseconds(
+    int h2_connection_keepalive_idle_interval_milliseconds) {
+  this->h2_connection_keepalive_idle_interval_milliseconds_ =
+      h2_connection_keepalive_idle_interval_milliseconds;
+  return *this;
+}
+
+EngineBuilder&
+EngineBuilder::addH2ConnectionKeepaliveTimeoutSeconds(int h2_connection_keepalive_timeout_seconds) {
+  this->h2_connection_keepalive_timeout_seconds_ = h2_connection_keepalive_timeout_seconds;
+  return *this;
+}
```

Comment:
I'm actually not certain we want to expose these values at this layer. If this ends up being our primary mechanism to detect zombie connections, overriding this configuration seems like a pretty major footgun for a naive user of the library.

Reply:
@goaway I wasn't aware that was your preference. I described that I was exposing all the way at the Engine API. What would your preference be? If we dont expose them at the engine builder layer then we wouldn't be able to experiment with settings internally at Lyft.

Reply:
going to set up default API in a subsequent PR. per offline discussion.